packages/storage-plugin/src/storage.plugin.ts

```diff
@@ -25,7 +25,8 @@ export class NgxsStoragePlugin implements NgxsPlugin {
 
         if (typeof val !== 'undefined' && val !== null) {
           try {
-            val = options.deserialize(val);
+            const previousState = isMaster ? state : getValue(state, key);
+            val = options.deserialize(val, key, previousState);
```

Comment:
Is the previous state still necesary as a parameter after the other changes below that merge the state (line 45)?

Reply:
You got me there, I did not see these changes. I think passing previousState is still necessary. Furthermore, I do not agree with line 49, because it's behavior is not something I would expect while in 'redux' mindset. Although I think my solution and solution from line 49 doesn't collide.